gems/aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb

```diff
@@ -93,6 +96,49 @@ def after_initialize(client)
         client.config.sigv4a_signing_region_set = region_set
       end
 
+      private
+
+      def initialize_default_endpoint(client)
+        client_module = Object.const_get(client.class.name.rpartition('::').first)
+        if client.config.respond_to?(:endpoint_provider) &&
```

Comment:
There's a lot of nesting here. Can we break this up a bit with either methods and/or guard clauses?